pkg/cri/server/service_linux.go

```diff
@@ -49,16 +47,29 @@ func (c *criService) initPlatform() error {
 		selinux.SetDisabled()
 	}
 
-	// Pod needs to attach to at least loopback network and a non host network,
-	// hence networkAttachCount is 2. If there are more network configs the
-	// pod will be attached to all the networks but we will only use the ip
-	// of the default network interface as the pod IP.
-	c.netPlugin, err = cni.New(cni.WithMinNetworkCount(networkAttachCount),
-		cni.WithPluginConfDir(c.config.NetworkPluginConfDir),
-		cni.WithPluginMaxConfNum(c.config.NetworkPluginMaxConfNum),
-		cni.WithPluginDir([]string{c.config.NetworkPluginBinDir}))
-	if err != nil {
-		return errors.Wrap(err, "failed to initialize cni")
+	pluginDirs := map[string]string{
+		defaultNetworkPlugin: c.config.NetworkPluginConfDir,
+	}
+	for name, conf := range c.config.Runtimes {
+		if conf.NetworkPluginConfDir != "" {
+			pluginDirs[name] = conf.NetworkPluginConfDir
+		}
+	}
+
+	c.netPlugin = make(map[string]cni.CNI)
+	for name, dir := range pluginDirs {
+		// Pod needs to attach to at least loopback network and a non host network,
+		// hence networkAttachCount is 2. If there are more network configs the
+		// pod will be attached to all the networks but we will only use the ip
+		// of the default network interface as the pod IP.
+		i, err := cni.New(cni.WithMinNetworkCount(networkAttachCount),
+			cni.WithPluginConfDir(dir),
+			cni.WithPluginMaxConfNum(c.config.NetworkPluginMaxConfNum),
```

Comment:
How about adding the `NetworkPluginMaxConfNum` based on runtime class? I think it is reasonable to configure conf number for specific runtime. (If possible, it can be follow-up :p)

Reply:
done :)